src/main/java/com/nulabinc/zxcvbn/Scoring.java

```diff
@@ -138,9 +140,10 @@ private static Match makeBruteforceMatch(CharSequence password, int i, int j) {
         return MatchFactory.createBruteforceMatch(i, j, password.subSequence(i, j + 1));
     }
 
-    private static int factorial(int n) {
+    private static long factorial(int n) {
         if (n < 2) return 1;
-        int f = 1;
+        if (n > 19) return JS_NUMBER_MAX;
```

Comment:
Notes:
n is 18                         =>     6402373705728000
JS_NUMBER_MAX    =>     9007199254740991
n is 19                          => 121645100408832000